default/python/universe_generation/empires.py

```diff
@@ -504,6 +567,24 @@ def setup_empire(empire, empire_name, home_system, starting_species, player_name
     print("Starting species for player", player_name, "is", starting_species)
     universe_statistics.empire_species[starting_species] += 1
 
+    # If game rule "RULE_ENSURE_HABITABLE_PLANET_HW_VICINITY" is set check for an adequate or better planet and
+    # if missing add one
+    if fo.getGameRules().getToggle("RULE_ENSURE_HABITABLE_PLANET_HW_VICINITY"):
+        systems_in_vicinity = fo.systems_within_jumps_unordered(HS_VICINITY_RANGE, [home_system])
+        planet_types_in_vicinity = list_planet_types_in_systems(systems_in_vicinity)
+        acceptable_planet_types = set()
+        for planet_type in planet_types:
+            # if this planet type is an adequate or good environment for the species, add it to the set of acceptable planet types
+            if (
+                fo.species_get_planet_environment(starting_species, planet_type) == fo.planetEnvironment.good
```

Comment:
That means that some users will get `planetEnvironment.good` and others `planetEnvironment.adequate`. This does not look fair.  Should it be the same planet type for everyone?

Reply:
fairness is hard to guarantee without it always being an average sized good on a yellow star (or GG for Sly). The narrow tolerance species don't have an adequate type for instance. I didn't have a great answer to the fairness question, so I just implemented a simple version and defaulted the option to off. I'm not convinced it matters for single player and multiplayer probably needs something trickier to be fair.